modules/ballerina-native/src/test/java/org/ballerinalang/nativeimpl/functions/DataTableTest.java

```diff
@@ -254,6 +255,42 @@ public void testGetArrayByIndex() {
         Assert.assertEquals(booleanArray.get(new BString("2")).booleanValue(), true);
     }
 
+    @Test(description = "Check date time operation")
+    public void testDateTime() {
+        BValue[] args = new BValue[3];
+        Calendar cal = Calendar.getInstance();
+        cal.clear();
+        cal.set(Calendar.HOUR, 14);
+        cal.set(Calendar.MINUTE, 15);
+        cal.set(Calendar.SECOND, 23);
+        long time = cal.getTimeInMillis();
+        args[0] = new BString(String.valueOf(time));
+
+        cal.clear();
+        cal.set(Calendar.YEAR, 2017);
+        cal.set(Calendar.MONTH, 5);
+        cal.set(Calendar.DAY_OF_MONTH, 23);
+        long date = cal.getTimeInMillis();
+        args[1] = new BString(String.valueOf(date));
+
+        cal.clear();
+        cal.set(Calendar.HOUR, 16);
+        cal.set(Calendar.MINUTE, 33);
+        cal.set(Calendar.SECOND, 55);
+        cal.set(Calendar.YEAR, 2017);
+        cal.set(Calendar.MONTH, 1);
+        cal.set(Calendar.DAY_OF_MONTH, 25);
+        long timestamp = cal.getTimeInMillis();
+        args[2] = new BString(String.valueOf(timestamp));
+
+        BValue[] returns = BLangFunctions.invoke(bLangProgram, "testDateTime", args);
+
+        Assert.assertEquals(returns.length, 3);
+        Assert.assertEquals(((BLong) returns[0]).longValue(), time);
```

Comment:
Underlaying code handle BLong at the moment. I can update https://github.com/ballerinalang/ballerina/pull/2018 once this merge.